lep/DRAFT/011_Performance_Monitor_Goals_and_Metrics.md

```diff
@@ -0,0 +1,85 @@
+#LEP 011 - Performance Monitor Goals and Metrics
+
+Date:   March 24, 2015
+
+Status: Draft
+
+Author: fffw
+
+## Abstract
+
+To continuously monitor the performance of Lantern network, we need to define dimensions and metrics clearly, and collect data to fulfill the goal at both given and get mode.
+
+We must also keep user's anonymity.
+
+## Proposal
+
+The output, is a multi-dimensional model which can pivot from any dimension. The presentation solution is TBD, hopefully there are existing services or open source projects.
+The fact to measure is data flow. For example, we can query:
+* Average time required to establish connection from Iran to Facebook last weekend
+* RTT from the given mode servers at each data center to Google
+* Total throughput from Iran to Amsterdam data center per day
+* Error rate comparison between beta version and previous release of Lantern in each country
+* ...
+
+###Terms
+*  **Ingress**: the node where request enters Lantern network
+*  **Egress**: the node where request exits Lantern network and hit the site
+
+###Dimensions
+* Ingress country/territory
+* Egress country/territory
+* Egress server pool, if that is the case
+* Ingress Lantern version
+* Egress Lantern version
+* Site to visit
+* Time of day (per half hour)
+* Day of year/month/week
```

Comment:
I would add:

- Technique (chained versus fronted)
- Port (which is a nice stand-in for protocol)